main/EDInitMod.F90

```diff
@@ -487,15 +590,50 @@ subroutine init_cohorts( site_in, patch_in, bc_in)
 
     patch_in%tallest  => null()
     patch_in%shortest => null()
-    
+
+    ! Manage interactions of ixed biogeg (site level filter) and 
+    ! nocomp (patch level filter) 
+    ! Need to cover all potential biogeog x nocomp combinations 
+    ! 1. biogeog = false. nocomp = false: all PFTs on (DEFAULT)
+    ! 2. biogeog = true.  nocomp = false: site level filter
+    ! 3. biogeog = false. nocomp = true : patch level filter
+    ! 4. biogeog = true.  nocomp = true : patch and site level filter
+    ! in principle this could be a patch level variable. 
+    do pft =  1,numpft 
+      ! Turn every PFT ON, unless we are in a special case.
+      use_pft_local(pft) = itrue ! Case 1
+      if(hlm_use_fixed_biogeog.eq.itrue)then !filter geographically
+        use_pft_local(pft) = site_in%use_this_pft(pft) ! Case 2
+        if(hlm_use_nocomp.eq.itrue.and.pft.ne.patch_in%nocomp_pft_label)then
+           ! Having set the biogeog filter as on or off, turn off all patches 
+           ! whose identiy does not correspond to this PFT. 
+           use_pft_local(pft) = ifalse ! Case 3
+        endif
+      else 
+        if(hlm_use_nocomp.eq.itrue.and.pft.ne.patch_in%nocomp_pft_label)then 
+        ! This case has all PFTs on their own patch everywhere. 
+           use_pft_local(pft) = ifalse ! Case 4 
+        endif
+      endif
+
+    end do
+      
     do pft =  1,numpft
-     if(site_in%use_this_pft(pft).eq.itrue)then
+     if(use_pft_local(pft).eq.itrue)then
        if(EDPftvarcon_inst%initd(pft)>1.0E-7) then
 
        allocate(temp_cohort) ! temporary cohort
 
        temp_cohort%pft         = pft
        temp_cohort%n           = EDPftvarcon_inst%initd(pft) * patch_in%area
+       if(hlm_use_nocomp.eq.itrue)then !in nocomp mode we only have one PFT per patch
```

Comment:
would it be simpler to define this on the line above as: 
`temp_cohort%n           = EDPftvarcon_inst%initd(pft) * AREA`
since, in the default case, there should only be one patch and thus patch_in%area would equal AREA, and then skip the logic below?

Reply:
I guess that should result in the same answer as AREA/cp%area = numpft

I'm a little afraid of just adding in the initd *  AREA statement though, as it is less clear what is going on.

Reply:
I've added a comment to the effect that this  is the same as initd* AREA